libs/block-editor/src/lib/extensions/imageUpload.extention.ts

```diff
@@ -0,0 +1,154 @@
+import { Plugin, PluginKey } from 'prosemirror-state';
+import { ComponentFactoryResolver, ComponentRef, Injector } from '@angular/core';
+import { Extension } from '@tiptap/core';
+import { DotImageService } from './services/dot-image/dot-image.service';
+import { EditorView } from 'prosemirror-view';
+import { MessageComponent, MessageType } from './components/message/message.component';
+import { PlaceholderPlugin } from '../plugins/placeholder.plugin';
+import { take } from 'rxjs/operators';
+
+export const ImageUploadExtension = (injector: Injector, resolver: ComponentFactoryResolver) => {
+    return Extension.create({
+        name: 'imageUpload',
+
+        addProseMirrorPlugins() {
+            const dotImageService = injector.get(DotImageService);
+            const messageBlockFactory = resolver.resolveComponentFactory(MessageComponent);
+
+            function areImageFiles(files: FileList): boolean {
+                for (let i = 0; i < files.length; i++) {
+                    if (!files[i].type.startsWith('image/')) {
+                        return false;
+                    }
+                }
+                return !!files.length;
```

Comment:
If we can only add one image at a time, why do we need to iterate here?

Reply:
this is from the event, it comes as a `FileList` even if is one